test/terra/backends/qasm_simulator/qasm_snapshot.py

```diff
@@ -484,3 +486,84 @@ def test_snapshot_density_matrix_post_measure_nondet(self):
                     target = np.outer(target, target.conj())
                     value = snaps.get(mem)
                     self.assertTrue(np.allclose(value, target))
+
+
+class QasmSnapshotProbabilitiesTests:
+    """QasmSimulator snapshot probabilities tests."""
+
+    SIMULATOR = QasmSimulator()
+    SUPPORTED_QASM_METHODS = [
+        'automatic', 'statevector', 'density_matrix', 'matrix_product_state'
+    ]
+    BACKEND_OPTS = {}
+
+    @staticmethod
+    def probabilitiy_snapshots(data, labels):
+        """Format snapshots as nested dicts"""
+        # Check snapshot entry exists in data
+        output = {}
+        for label in labels:
+            snaps = data.get("snapshots", {}).get("probabilities", {}).get(label, [])
+            # Convert list into dict
+            inner = {}
+            for snap_dict in snaps:
+                inner[snap_dict['memory']] = snap_dict['value']
```

Comment:
Nit: I was wondering how to write this using comprehesion dictionaries, but I don't like the result, what do you think?
```python
inner = { k: v for m,k in snaps.items() if m == "memory" for m,v in snaps.items() if  m == "value" }
```

Reply:
hahahaha ok, way better (thanks @delapuente)
```python
inner = { snap_dict['memory']: snap_dict['value'] for snap_dict in snaps }
```

Reply:
... there are days, you know..